src/main/java/be/syntra/devshop/DevshopFront/controllers/SearchController.java

```diff
@@ -72,69 +74,68 @@ public String searchProductDescription(@RequestParam String description, Model m
+        searchService.setArchivedView(true);
+        List<Product> productList = productService.findAllProductsBySearchModel().getProducts();
+        return setTemplateModel(model, productList);
     }
 
     @GetMapping("/archived/sortbyprice")
     public String sortArchivedByPrice(Model model) {
-        List<Product> productList = productService.findAllArchived().getProducts();
-        return getProductsSortedByPrice(model, productList);
+        disableNameSorting();
+        reversePriceSorting();
+        searchService.setArchivedView(true);
+        List<Product> productList = productService.findAllProductsBySearchModel().getProducts();
+        return setTemplateModel(model, productList);
     }
 
     private List<Product> applySearch(boolean isSearchResultView) {
         searchService.setSearchResultView(isSearchResultView);
         searchService.setArchivedView(false);
-        List<Product> productList = productListCacheService.findBySearchRequest(searchService.getSearchModel()).getProducts();
-        List<Product> filteredList = productListCacheService.filterByPrice(productList, searchService.getSearchModel()).getProducts();
-        return hasDescription()
-                ? productListCacheService.searchForProductDescription(filteredList, searchService.getSearchModel()).getProducts()
-                : filteredList;
+        return productService.findAllProductsBySearchModel().getProducts();
     }
 
-    private boolean hasDescription() {
-        return null != searchService.getSearchModel().getDescription();
-    }
-
-    private String getProductsSortedByName(Model model, List<Product> productList) {
-        List<Product> sortedProducts = productListCacheService.sortListByName(productList, searchService.getSearchModel()).getProducts();
-        reverseNameSort();
-        return setTemplateModel(model, sortedProducts);
-    }
-
-    private String getProductsSortedByPrice(Model model, List<Product> productList) {
-        List<Product> sortedProducts = productListCacheService.sortListByPrice(productList, searchService.getSearchModel()).getProducts();
-        reversePriceSort();
-        return setTemplateModel(model, sortedProducts);
-    }
-
-    private void reverseNameSort() {
+    private void reverseNameSorting() {
         boolean sortAscending = searchService.getSearchModel().isSortAscendingName();
         searchService.getSearchModel().setSortAscendingName(!sortAscending);
     }
 
-    private void reversePriceSort() {
+    private void reversePriceSorting() {
         boolean sortAscending = searchService.getSearchModel().isSortAscendingPrice();
         searchService.getSearchModel().setSortAscendingPrice(!sortAscending);
     }
 
     private String setTemplateModel(Model model, List<Product> productList) {
+        ProductListDto productListDto = new ProductListDto(productList);
+        ProductDtoList productDtoListDto = productMapperUtil.convertToProductDtoList(productListDto);
```

Comment:
Slightly confused about a `ProductListDto` and `ProductDtoList`. The naming seems so similar that one could be redundant?

Reply:
Will change

Reply:
what Steve said. If you just kept the ProductList (which is a Dto), you could write: 
ProductList products = new ProductList(productList). 
Since this is the only place where you're using it, the productMapperUtil could be removed entirely from the class

Reply:
This is the object that contains the Products for the main list of products on the homepage, it includes the count in the cart for each product, that count is set in the mapper.  I can name it **ProductsDisplayList** , i made this object to have a separate object with products that have the count copied into them from the cart.